### DISCUSSION TITLE
Traits in V5 ?

### DISCUSSION THREAD

**Discussion by hsharma1996** [adonisjs/core#3188] [Help] [answered]
Any suggestion on sharing common function between models ? 
The extend class method does not work. 
Is there any other way we can achieve it ?

**Comment by thetutlage** [comment, marked as answer]
You cannot make traits work with TypeScript. The TypeScript recommended approach is to make use of Mixins. I suggest you to read about Mixins in general and then read this AdonisJS doc for the [compose](https://docs.adonisjs.com/guides/helpers#compose). Compose method just makes the syntax a little better for mixins
